google/cloud/bigquery/table.py

```diff
@@ -451,6 +453,15 @@ def schema(self):
                 If ``schema`` is not a sequence, or if any item in the sequence
                 is not a :class:`~google.cloud.bigquery.schema.SchemaField`
                 instance or a compatible mapping representation of the field.
+
+        NOTE: If you are referencing a schema for an external catalog table such
+        as a Hive table, it will also be necessary to populate the foreign_type_info
+        attribute. This is not necessary if defining the schema for a BigQuery table.
+
+        For details, see:
+        https://cloud.google.com/bigquery/docs/external-tables
+        https://cloud.google.com/bigquery/docs/datasets-intro#external_datasets
+
```

Comment:
I think we'll need some logic in the setter for schema to avoid overwriting the `schema` property entirely. Instead, it'll need to be responsible for just `schema.fields`.

Reply:
Also, I'm not sure if this format will render well in the docs. We might just move all the contents under `NOTE:` to after `Table's schema.`

Reply:
I have not yet addressed Tim's comment here.
I spoke with Linchin about the note and with the revision I added, Sphinx should be able to handle the note with no problem.